app/src/org/storymaker/app/server/ServerManager.java

```diff
@@ -124,7 +125,11 @@ public void connect (String username, String password) throws IOException // Mal
         // wrapper now checks tor settings
         try {
             smWrapper.login(username, password);
-        } catch (IOException ioe) {
+        } catch (CaptchaException ce) {
+			Log.e(TAG, "connect() bailing out, server returned captcha challenge");
+			// throw exception so LoginActivity doesn't save credentials
+			throw new IOException("Some TOR IPs may be flagged as suspicious, try restarting TOR");
```

Comment:
that's the IOException from the existing code, it catches generic login failures